File: mgyoutube-multitech/api-dotnet/repos/SearchesDataRepo.cs
using System.Collections.Generic;

namespace api_dotnet.repos
{
    public interface SearchesDataRepo
    {
        void RepositoryStartup();

        List<string> GetSearchesForParentUser(string userId);

        void AddSearchToParentUser(string parentUserId, string searchPhrase);
    }
}

File: mgyoutube-multitech/api-go/src/repos/searchesdatarepo.go
package repos

// SearchesDataRepo - interface for search data repositories
type SearchesDataRepo interface {
	RepositoryStartup() error

	GetSearchesForParentUser(userID string) []string

	AddSearchToParentUser(parentUserID string, searchPhrase string)

	RemoveSearchFromParent(parentUserID string, searchPhrase string)
}
